types/react-bootstrap-table/index.d.ts

```diff
@@ -203,545 +498,2148 @@ export interface SelectRow {
 	 */
-	onSelectAll?: (isSelected: boolean, currentSelectedAndDisplayData: any) => boolean;
-
+	customComponent?(props: CustomSelectProps): string | ReactElement<any>;
 	/**
-	 * Provide a list of unselectable row keys.
+	 * Only unselect visible rows.
 	 */
-	unselectable?: number[];
+	onlyUnselectVisible?: boolean;
 }
 
-export type CellEditClickMode = 'none' | 'click' | 'dbclick';
-
+/**
+ * react-bootstrap-table supports cell editing. When you enable this feature, react-bootstrap-table will make
+ * the target cell editable by either clicking or dbclicking (depending on the properties you set).
+ */
 export interface CellEdit {
 	/**
-	 * To spectify which condition will trigger cell editing.(click or dbclick)
+	 * Spectify which condition will trigger cell editing.(click or dbclick). Required.
 	 */
 	mode: CellEditClickMode;
 	/**
-	 * Enable blurToSave will trigger a saving event on cell when mouse blur on the input field. Default is false.
+	 * Enabling blurToSave will trigger a saving event on the cell when the input field becomes deselected. Default is false.
 	 * In the default condition, you need to press ENTER to save the cell.
 	 */
 	blurToSave?: boolean;
 	/**
-	 * Accept a custom callback function, before cell saving, this function will be called.
-	 * This callback function taking three arguments:row, cellName and cellValue
-	 * It's necessary to return a bool value which whether apply this cell editing.
+	 * Enabling blurToEscape will result in a cell edit being cancelled when the user clicks outside the table during
+	 * editing.
+	 * Default is false.
+	 */
+	blurToEscape?: boolean;
+	/**
+	 * nonEditableRows tell react-bootstrap-table which rows should not be edited on all of the columns. Briefly, its a row level limitation
+	 * Please assign a callback function, and this function is supposed to be return an array of row keys.
 	 */
-	beforeSaveCell?: (row: any, cellName: string, cellValue: any) => boolean;
+	nonEditableRows?(): Array<RowKey<any>>;
+	/**
+	 * Accept a custom callback function, before cell saving, this function will be called.
+	 * This callback function takes four arguments: row, cellName, cellValue and done.
+	 *   `row`: the row data to be saved.
+	 *   `cellName`: the column dataField cell name that has been modified.
+	 *   `cellValue`: the new cell value.
+	 *   `done`: a callback function to use if this is an async operation, to indicate if the save data is valid.
+	 * If your validation is async, for example: you want to pop a confirm dialog for user to confim in this case,
+	 * react-bootstrap-table pass a callback function to you. You are supposed to call this callback function with a
+	 * bool value to perfom if it is valid or not in addition, you should return 1 from the main function to tell
+	 * react-bootstrap-table that this is a async operation.
+	 */
+	beforeSaveCell?(row: Row<any>, cellName: string, cellValue: Cell<any>, done: (isValid: boolean) => void): boolean | 1;
```

Comment:
Suggestion (not required for approval): If you make `CellEdit` generic over a given row (e.g. `interface CellEdit<TRow extends object = any>`), you could make `beforeSaveCell` more typesafe by defining it as:

```ts
  beforeSaveCell?<K extends keyof TRow>(row: TRow, cellName: K, cellValue: TRow[K], done: (isValid: boolean) => void): boolean | 1;
```